widgets/pie/pie.go

```diff
@@ -0,0 +1,167 @@
+package pie
+
+import (
+	"errors"
+	"fmt"
+	"image"
+	"math"
+	"sync"
+
+	"github.com/mum4k/termdash/cell"
+	"github.com/mum4k/termdash/private/canvas"
+	"github.com/mum4k/termdash/private/canvas/braille"
+	"github.com/mum4k/termdash/private/draw"
+	"github.com/mum4k/termdash/terminal/terminalapi"
+	"github.com/mum4k/termdash/widgetapi"
+)
+
+// Pie is the widget that displays a pie chart.
+type Pie struct {
+	mu     sync.Mutex
+	values []int
+	total  int
+	colors []cell.Color
+	opts   *options
+}
+
+// New returns a new Pie widget.
+func New(opts ...Option) (*Pie, error) {
+	opt := newOptions()
+	for _, o := range opts {
+		o.set(opt)
+	}
+	return &Pie{
+		opts: opt,
+	}, nil
+}
+
+// Values must be provided before calling Draw.
+func (p *Pie) Values(values []int, opts ...Option) error {
+	// The values must be non-negative and a color must be provided for each value.
+	// If not enough colors are provided, they will be reused.
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	if len(values) == 0 {
+		return errors.New("values cannot be empty")
+	}
+
+	for _, opt := range opts {
+		opt.set(p.opts)
+	}
+
+	p.values = values
+	p.total = 0
+	if len(p.colors) == 0 {
+		p.colors = DefaultColors
```

Comment:
Typically this should be handled by the initialization of options in the options.go file. It might be redundant here, though it doesn't hurt.